code/lib/store/template/stories/rendering.stories.ts

```diff
@@ -40,21 +45,25 @@ export const ForceRemount = {
 export const ChangeArgs = {
   play: async ({ canvasElement, id }: PlayFunctionContext<any>) => {
     const channel = globalThis.__STORYBOOK_ADDONS_CHANNEL__;
+
+    await channel.emit(RESET_STORY_ARGS, { storyId: id });
+    await new Promise((resolve) => {
+      channel.once(STORY_ARGS_UPDATED, resolve);
+    });
+
     const button = await within(canvasElement).findByRole('button');
     await button.focus();
     await expect(button).toHaveFocus();
 
     // Web-components: https://github.com/storybookjs/storybook/issues/19415
     // Preact: https://github.com/storybookjs/storybook/issues/19504
+
     if (['web-components', 'html', 'preact'].includes(globalThis.storybookRenderer)) return;
 
     // When we change the args to the button, it should not remount
     await channel.emit(UPDATE_STORY_ARGS, { storyId: id, updatedArgs: { label: 'New Text' } });
-    await within(canvasElement).findByText(/New Text/);
-    await expect(button).toHaveFocus();
 
-    await channel.emit(RESET_STORY_ARGS, { storyId: id });
```

Comment:
Can we test the reset functionality as well? Possibly in a separate story if you want to test the changing of args in this one?

Reply:
i just move it up us it is kind of standard why any other tests